File: src/main/java/de/pbauerochse/worklogviewer/fx/tablecolumns/CellStyleClasses.java
package de.pbauerochse.worklogviewer.fx.tablecolumns;

import java.util.HashSet;
import java.util.Set;

/**
 * @author Patrick Bauerochse
 * @since 08.07.15
 */
public class CellStyleClasses {

    public static final String SUMMARY_COLUMN_OR_CELL_CSS_CLASS = "summary";
    public static final String GROUP_COLUMN_OR_CELL_CSS_CLASS = "group";
    public static final String HIGHLIGHT_COLUMN_CSS_CLASS = "highlight";
    public static final String TODAY_COLUMN_OR_CELL_CSS_CLASS = "today";
    public static final String ISSUE_CELL_CSS_CLASS = "issue-cell";

    public static final Set<String> ALL_WORKLOGVIEWER_CLASSES = new HashSet<>();
    static {
        ALL_WORKLOGVIEWER_CLASSES.add(SUMMARY_COLUMN_OR_CELL_CSS_CLASS);
        ALL_WORKLOGVIEWER_CLASSES.add(GROUP_COLUMN_OR_CELL_CSS_CLASS);
        ALL_WORKLOGVIEWER_CLASSES.add(HIGHLIGHT_COLUMN_CSS_CLASS);
        ALL_WORKLOGVIEWER_CLASSES.add(TODAY_COLUMN_OR_CELL_CSS_CLASS);
        ALL_WORKLOGVIEWER_CLASSES.add(ISSUE_CELL_CSS_CLASS);
    }

}

File: src/main/java/de/pbauerochse/worklogviewer/fx/tablecolumns/CellStyleClasses.kt
package de.pbauerochse.worklogviewer.fx.tablecolumns

/**
 * Defines all used CSS classes to style
 * the components in the worklog viewer
 */
object CellStyleClasses {

    const val SUMMARY_CELL = "summary-cell"
    const val TIMESPENT_CELL = "timespent-cell"
    const val GROUP_CELL = "group-cell"
    const val HIGHLIGHT_CELL = "highlight-cell"
    const val TODAY_HIGHLIGHT_CELL = "today-highlight-cell"
    const val ISSUE_LINK_CELL = "issue-cell"

    val ALL_WORKLOGVIEWER_CLASSES: Set<String> = setOf(
        SUMMARY_CELL,
        GROUP_CELL,
        HIGHLIGHT_CELL,
        TODAY_HIGHLIGHT_CELL,
        ISSUE_LINK_CELL,
        TIMESPENT_CELL
    )

}
